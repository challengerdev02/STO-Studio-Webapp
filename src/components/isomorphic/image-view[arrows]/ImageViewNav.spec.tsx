import {
  cleanup,
  fireEvent,
  queryAllByAltText,
  queryAllByRole,
  queryByAttribute,
  queryByRole,
  render,
  waitFor,
} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { ImageViewNavComponent } from './index';

describe('ImageViewNavComponent', function () {
  afterEach(cleanup);
  let props = {
    images: [
      {
        src: 'https://i.picsum.photos/id/111/600/400.jpg?hmac=hVzgLIvxNZYaZkGV8HgOs6FO0aePt5tSEhb-sZBwbkk',
      },
      {
        src: 'https://i.picsum.photos/id/848/600/400.jpg?hmac=gdx1i308nOCrjtfzwAlWF5RCN8HKoQd-POMetrAmZxc',
      },
    ],
  };

  const setup = (extraProps: Record<string, any> = {}) => {
    const utils = render(<ImageViewNavComponent {...props} {...extraProps} />);
    return {
      ...utils,
    };
  };

  beforeEach(() => {
    Object.defineProperty(window, 'matchMedia', {
      writable: true,
      value: jest.fn().mockImplementation((query) => ({
        matches: false,
        media: query,
        onchange: null,
        addListener: jest.fn(), // deprecated
        removeListener: jest.fn(), // deprecated
        addEventListener: jest.fn(),
        removeEventListener: jest.fn(),
        dispatchEvent: jest.fn(),
      })),
    });
  });

  it('should match with snapshot', () => {
    // @ts-ignore
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });

  it('should display with NO image container', () => {
    // @ts-ignore
    const { container } = setup({ images: [] });
    expect(queryAllByRole(document.body, 'main-image-view').length).toBe(0);
  });

  it('should display the main image container', () => {
    // @ts-ignore
    const { container } = setup();
    expect(
      queryAllByRole(document.body, 'main-image-view')[0]
    ).toBeInTheDocument();
    expect(queryAllByRole(document.body, 'main-image-view')[0]).toHaveClass(
      'main-image-view'
    );
  });

  it('should display the Nav container', () => {
    // @ts-ignore
    const { container } = setup();
    expect(queryByRole(document.body, 'nav-container')).toBeInTheDocument();
  });

  it('should have Images', () => {
    // @ts-ignore
    const {} = setup();
    expect(queryAllByAltText(document.body, 'image')[0]).toBeInTheDocument();
    expect(queryAllByAltText(document.body, 'image')[0]).toHaveAttribute(
      'src',
      'https://i.picsum.photos/id/111/600/400.jpg?hmac=hVzgLIvxNZYaZkGV8HgOs6FO0aePt5tSEhb-sZBwbkk'
    );
  });

  it('should have a expand button', () => {
    // @ts-ignore
    const { container } = setup();
    expect(
      queryByAttribute('class', document.body, 'ant-image-mask')
    ).toBeInTheDocument();
  });

  it('should fire onCancel callback when modal is closed', async () => {
    // @ts-ignore
    const { container } = setup();

    await waitFor(() => {
      const btn = queryByAttribute('class', document.body, 'ant-image-mask');

      if (btn) {
        fireEvent.click(btn);
      }
    });

    await waitFor(() => {
      expect(queryByRole(document.body, 'dialog')).toBeInTheDocument();
      expect(queryByRole(document.body, 'dialog')).not.toHaveStyle(
        'display: none'
      );
    });
  });

  it('should fire Next callback Next button is clicked', async () => {
    // @ts-ignore
    const { container } = setup();

    await waitFor(() => {
      const btn = queryByRole(document.body, 'next-slide-btn');
      if (btn) {
        fireEvent.click(btn);
      }
    });

    await waitFor(() => {
      expect(queryByRole(document.body, 'current-slide')).toBeInTheDocument();
      expect(queryByRole(document.body, 'current-slide')).toHaveTextContent(
        '2'
      );
    });

    await waitFor(() => {
      const btn = queryByRole(document.body, 'previous-slide-btn');
      if (btn) {
        fireEvent.click(btn);
      }
    });

    await waitFor(() => {
      expect(queryByRole(document.body, 'current-slide')).toBeInTheDocument();
      expect(queryByRole(document.body, 'current-slide')).toHaveTextContent(
        '1'
      );
    });
  });

  it('should fire Previous callback Previous button is clicked', async () => {
    // @ts-ignore
    const { container } = setup();

    await waitFor(() => {
      const btn = queryByRole(document.body, 'previous-slide-btn');
      if (btn) {
        fireEvent.click(btn);
      }
    });

    await waitFor(() => {
      expect(queryByRole(document.body, 'current-slide')).toBeInTheDocument();
      expect(queryByRole(document.body, 'current-slide')).toHaveTextContent(
        '2'
      );
    });

    await waitFor(() => {
      const btn = queryByRole(document.body, 'next-slide-btn');
      if (btn) {
        fireEvent.click(btn);
      }
    });

    await waitFor(() => {
      expect(queryByRole(document.body, 'current-slide')).toBeInTheDocument();
      expect(queryByRole(document.body, 'current-slide')).toHaveTextContent(
        '1'
      );
    });
  });
});
