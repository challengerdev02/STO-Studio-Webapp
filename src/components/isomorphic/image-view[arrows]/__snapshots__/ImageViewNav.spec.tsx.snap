// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageViewNavComponent should match with snapshot 1`] = `
<div>
  <div
    class="sc-gsDKAQ gwlFDG image-view-container"
  >
    <div
      class="sc-bdvvtL oKzls image-view-container main-image-view"
      role="main-image-view"
    >
      <div
        class="ant-image"
        role="main-image-view"
      >
        <img
          alt="image"
          class="ant-image-img main-image-view"
          src="https://i.picsum.photos/id/111/600/400.jpg?hmac=hVzgLIvxNZYaZkGV8HgOs6FO0aePt5tSEhb-sZBwbkk"
        />
        <div
          class="ant-image-mask"
        >
          <span
            class="expand-arrow"
          >
            <svg
              fill="none"
              height="25"
              viewBox="0 0 26 25"
              width="26"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M15.988 8.36399L21.1585 3.42762C21.4671 3.13296 21.3659 2.89352 20.9277 2.89671L15.5575 2.936C13.9552 2.94821 13.9735 0.653066 15.5758 0.641378L24.7061 0.500149C25.3601 0.490062 25.8935 0.993365 25.8851 1.62514L25.7633 10.3665C25.7511 11.8961 23.3474 11.9142 23.3596 10.3841L23.4013 5.25755C23.4047 4.83388 23.1555 4.74097 22.8452 5.03722L10.7023 16.6291C10.6951 16.636 10.6884 16.6424 10.6811 16.6488L5.51013 21.5851C5.20146 21.8798 5.30269 22.1192 5.74093 22.1161L11.1117 22.0762C12.714 22.0646 12.695 24.3592 11.0928 24.3714L1.94912 24.4999C1.29397 24.5089 0.760624 24.0056 0.770647 23.3749L0.905235 14.6461C0.91747 13.1166 3.32118 13.0985 3.30893 14.6281L3.26722 19.7546C3.26388 20.1788 3.51304 20.2717 3.82336 19.9749L15.9669 8.38355C15.9735 8.37665 15.9808 8.37028 15.988 8.36391L15.988 8.36399Z"
                fill="black"
                fill-rule="evenodd"
              />
            </svg>
          </span>
        </div>
      </div>
    </div>
    <div
      class="sc-hKwDye bCucgT"
      role="nav-container"
    >
      <button
        class="ant-btn ant-btn-text"
        role="previous-slide-btn"
        type="button"
      >
        <i
          class="mc-arrow-left-2-line mc-2x"
        />
      </button>
      <span
        class="ant-typography"
      >
        <span
          class="current"
          role="current-slide"
        >
          1
        </span>
        <span>
          /
        </span>
        <span
          class="total"
        >
          2
        </span>
      </span>
      <button
        class="ant-btn ant-btn-text"
        role="next-slide-btn"
        type="button"
      >
        <i
          class="mc-arrow-right-2-line mc-2x"
        />
      </button>
    </div>
  </div>
</div>
`;
